src/daemon/daemon.cpp

```diff
@@ -278,23 +277,40 @@ auto fetch_image_for(const std::string& name, const mp::FetchType& fetch_type, m
     return vault.fetch_image(fetch_type, query, stub_prepare, stub_progress);
 }
 
+auto try_mem_size(const std::string& val) -> mp::optional<mp::MemorySize>
+{
+    try
+    {
+        return mp::MemorySize{val};
+    }
+    catch (mp::InvalidMemorySizeException& /*unused*/)
+    {
+        return mp::nullopt;
+    }
+}
+
 auto validate_create_arguments(const mp::LaunchRequest* request)
 {
-    auto mem_size = request->mem_size();
-    auto disk_space = request->disk_space();
+    static const auto min_mem = try_mem_size(mp::min_memory_size);
+    static const auto min_disk = try_mem_size(mp::min_disk_size);
+    assert(min_mem && min_disk);
+
+    auto mem_size_str = request->mem_size();
+    auto disk_space_str = request->disk_space();
     auto instance_name = request->instance_name();
     auto option_errors = mp::LaunchError{};
 
-    const auto opt_mem_size = mp::utils::in_bytes(mem_size.empty() ? "1G" : mem_size);
-    const auto opt_disk_space = mp::utils::in_bytes(disk_space.empty() ? "5G" : disk_space);
+    const auto opt_mem_size = try_mem_size(mem_size_str.empty() ? "1G" : mem_size_str);
+    const auto opt_disk_space = try_mem_size(disk_space_str.empty() ? "5G" : disk_space_str);
```

Comment:
IIUC this silently accepts invalid entries and "converts" them into the defaults? Should this not be an error instead?

Reply:
I thought the RPC parameter would be empty only when the user did not provide it, in which case we want the default, no? Anyway, I believe this comes directly from transforming what we had before, but I will have a better look.

Reply:
Sorry, just read properly below, it does error out there.